lib/Doctrine/ODM/MongoDB/Query/ReferencePrimer.php

```diff
@@ -232,6 +226,8 @@ private function parseDotSyntaxForPrimer(string $fieldName, ClassMetadata $class
 
             return ['fieldName' => $fieldName, 'class' => $class, 'documents' => $documents, 'mapping' => $mapping];
         }
+
+        return ['fieldName' => $fieldName, 'class' => $class, 'documents' => $documents, 'mapping' => $mapping];
```

Comment:
I have not knowledge enough about this, I've just added the same line than https://github.com/doctrine/mongodb-odm/blob/6e373600959a6a50296724995b2315c51563071f/lib/Doctrine/ODM/MongoDB/Query/ReferencePrimer.php#L171.

Reply:
You're right, an exception would be better if this should never be reached. This should read just fine:
```suggestion
        throw new \LogicException('Unable to parse property path for ReferencePrimer. Please report an issue in Doctrine\'s MongoDB ODM.');
```